File: solutions/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit_max=0
        min=prices[0]
        for i in range(1,len(prices)):
            profit=prices[i]-min
            if profit>profit_max:
                profit_max=profit
            
            if prices[i]<min:
                min=prices[i]
        return profit_max
            
        

File: solutions/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
     //    int min=prices[0];
     //    int profit=0;
     //    for (int i:prices) {
     //        profit=Math.max(profit,(i-min));
     //        if (i<min) {min=i;}
     //    }
     // return profit;
    
     int min=prices[0];
     int profit=0;
     for (int i: prices) {
         profit=Math.max(profit,i-min);
         if (i<min) {
             min=i;
         }    
     }
    return profit;                      
    }
}
